app/actions/authActions.js

```diff
@@ -77,14 +79,18 @@ export const nep2LoginActions = createActions(
 
 export const ledgerLoginActions = createActions(
   ID,
-  ({ publicKey }: LedgerLoginProps) => (): AccountType => {
+  ({ publicKey, account }: LedgerLoginProps) => (): AccountType => {
     const publicKeyEncoded = wallet.getPublicKeyEncoded(publicKey)
-    const account = new wallet.Account(publicKeyEncoded)
+    const walletAccount = new wallet.Account(publicKeyEncoded)
 
     return {
       publicKey,
-      address: account.address,
-      signingFunction: ledgerNanoSCreateSignatureAsync,
+      address: walletAccount.address,
+      signingFunction: bindArgsFromN(
```

Comment:
what are you trying to achieve here? I'm a bit confused by the `bindArgsFromN` function

Reply:
So I probably should have left a comment here but binding the account (3rd argument only) to the signingFunc is the only way that this whole process works with the way things are currently set up.

Reply:
Previously nothing was passed in so the logic always defaulted to `account = 0`

Reply:
the signing function is for the ledger. Basically it never shares the private key. What happens is that when the user is using a hardware device, the transaction is passed to the ledger in a message, when the user clicks the button, it signs the transaction and sends the signed message back to the application, thereby never revealing the private key that was used to sign the transaction. I am not sure why we are binding anything, but just wanted to clarify how that works in case it is helpful. Please let me know if you would like me to look into this code/ binding logic further

Reply:
@dvd-schwrtz like I mentioned before we have to bind the account otherwise the transaction will always default to account 0 (the first account) => https://github.com/CityOfZion/neon-wallet/blob/ff340cc636eca1165c056cddb86031794a1cd129/app/ledger/neonLedger.js#L264

Reply:
I am binding the third argument (account) when the user authenticates to prevent this...

Reply:
cool. lets change bindArgsFromN to bindActiveAccount or something

Reply:
The problem with calling it `bindActiveAccount` is that thats not what its doing... `bindArgsFromN` will bind any amount of function args after a specific argument number (N)